src/ert/config/ert_config.py

```diff
@@ -530,53 +609,70 @@ def filter_env_dict(self, d):
         def handle_default(fm_step: ForwardModelStep, arg: str) -> str:
             return fm_step.default_mapping.get(arg, arg)
 
-        for fm_step in self.forward_model_steps:
+        for fm_step in forward_model_steps:
             for key, val in fm_step.private_args.items():
                 if key in context and key != val and context[key] != val:
                     logger.info(
                         f"Private arg '{key}':'{val}' chosen over"
                         f" global '{context[key]}' in forward model step {fm_step.name}"
                     )
         config_file_path = (
-            Path(self.user_config_file) if self.user_config_file is not None else None
+            Path(user_config_file) if user_config_file is not None else None
         )
         config_path = str(config_file_path.parent) if config_file_path else ""
         config_file = str(config_file_path.name) if config_file_path else ""
+
+        job_list_errors = []
+        job_list: List[ForwardModelStepJSON] = []
+        for idx, fm_step in enumerate(forward_model_steps):
+            substituter = Substituter(fm_step)
+            fm_step_json = {
+                "name": substituter.substitute(fm_step.name),
+                "executable": substituter.substitute(fm_step.executable),
+                "target_file": substituter.substitute(fm_step.target_file),
+                "error_file": substituter.substitute(fm_step.error_file),
+                "start_file": substituter.substitute(fm_step.start_file),
+                "stdout": (
+                    substituter.substitute(fm_step.stdout_file) + f".{idx}"
+                    if fm_step.stdout_file
+                    else None
+                ),
+                "stderr": (
+                    substituter.substitute(fm_step.stderr_file) + f".{idx}"
+                    if fm_step.stderr_file
+                    else None
+                ),
+                "stdin": substituter.substitute(fm_step.stdin_file),
+                "argList": [
+                    handle_default(fm_step, substituter.substitute(arg))
+                    for arg in fm_step.arglist
+                ],
+                "environment": substituter.filter_env_dict(fm_step.environment),
+                "exec_env": substituter.filter_env_dict(fm_step.exec_env),
+                "max_running_minutes": fm_step.max_running_minutes,
+            }
+
+            try:
+                if not skip_pre_experiment_validation:
+                    fm_step_json = fm_step.validate_pre_realization_run(fm_step_json)
+            except BaseException as exc:
```

Comment:
Instead of `BaseException` I think we should create a specific exception that forward model validation can raise to indicate failed validation. This is in order to avoid catching things like keyboard interrupt (Ctrl-C) or AttributeError (something is wrong with the plugin implementation):
```
class InvalidForwardModelCall(Exception):
    """ Thrown when the user calls the forward model incorrectly.
          
          Can be subtyped by the implementation of ForwardModelStepPlugin and
          thrown from `validate_pre_realization_run` or `validate_pre_experiment`.
    """

Reply:
I added it to ert, called it `ForwardModelInvalidStepCall`, if you insist I will go with  `InvalidForwardModelCall` but it is nice to be able to just type `Fo..`, in the IDE and get up all the forwardmodel stuff on autocomplete.

Reply:
:+1: You also need to just catch that exception when you call `validate_pre_realization_run`.